src/keyword.rs

```diff
@@ -0,0 +1,25 @@
+use regex::Regex;
+
+pub fn restore(name: &String, keywords: &Vec<String>) -> String {
+    let keyword_list = to_regex(keywords);
+
+    let matches = keyword_list.iter().fold(false, |acc, keyword_regex| {
+        acc || keyword_regex.is_match(name)
+    });
+
+    if matches {
+        name[..name.len() - 1].to_string()
+    } else {
+        name.clone()
+    }
+}
+
+pub fn to_regex(keywords: &Vec<String>) -> Vec<Regex> {
```

Comment:
`to_regex` to me indicates that I will get one regex out. Consider changing this to `to_regex_list` (or something), or doing the mapping outside the function.